macos/FluentUI/Button/Button.swift

```diff
@@ -175,304 +271,355 @@ open class Button: NSButton {
+				borderColor = .clear
+			case .borderless:
+				contentTintColor = accentColor
+				contentTintColorDisabled = ButtonColor.brandForegroundDisabled
+				backgroundColor = .clear
+				borderColor = .clear
+			case .none:
 				break
 			@unknown default:
 				break
 			}
+		}
+	}
 
-			return NSMakeRect(x, y, width, height)
+	private var fontSize: CGFloat = 15 {
+		willSet {
+			font = NSFont.systemFont(ofSize:newValue)
 		}
+	}
 
-		override func titleRect(forBounds rect: NSRect) -> NSRect {
-			let isRTL = NSApp.userInterfaceLayoutDirection == .rightToLeft
-			let titleRect = super.titleRect(forBounds: rect);
-			let titleSize = title.size(withAttributes: [.font: font as Any])
-			
-			var x = titleRect.origin.x
-			var y = titleRect.origin.y
-			var width = titleRect.size.width
-			var height = titleRect.size.height
-			
-			let imageSize = image?.size ?? NSZeroSize
-						
-			switch imagePosition {
-			case .noImage:
-				x = Button.horizontalEdgePadding
-				y = Button.verticalEdgePadding
-				width = rect.width - (Button.horizontalEdgePadding * 2)
-				height = rect.height - (Button.verticalEdgePadding * 2)
-				break
-			case .imageOnly:
-				x = 0
-				y = 0
-				width = 0
-				height = 0
-				break
-			case .imageLeft:
-				x = Button.horizontalEdgePadding + imageSize.width + Button.titleAndImageRectInterspacing
-				y = Button.verticalEdgePadding
-				width = titleSize.width
-				height = titleSize.height
-				break
-			case .imageRight:
-				x = Button.horizontalEdgePadding
-				y = Button.verticalEdgePadding
-				width = titleSize.width
-				height = titleSize.height
-				break
-			case .imageBelow:
-				x = (rect.width - titleSize.width) / 2
-				y = Button.verticalEdgePadding
-				width =	titleSize.width
-				height = titleSize.height
-				break
-			case .imageAbove:
-				x = (rect.width - titleSize.width) / 2
-				y = Button.verticalEdgePadding + imageSize.height + Button.titleAndImageRectInterspacing
-				width =	titleSize.width
-				height = titleSize.height
-				break
-			case .imageOverlaps:
-				x = Button.horizontalEdgePadding
-				y = Button.verticalEdgePadding
-				width = rect.width - (Button.horizontalEdgePadding * 2)
-				height = rect.height - (Button.verticalEdgePadding * 2)
-				break;
-			case .imageLeading:
-				x = isRTL ? Button.horizontalEdgePadding : Button.horizontalEdgePadding + imageSize.width + Button.titleAndImageRectInterspacing
-				y = Button.verticalEdgePadding
-				width = titleSize.width
-				height = titleSize.height
-				break
-			case .imageTrailing:
-				x = isRTL ? Button.horizontalEdgePadding + imageSize.width + Button.titleAndImageRectInterspacing : Button.horizontalEdgePadding
-				y = Button.verticalEdgePadding
-				width = titleSize.width
-				height = titleSize.height
-				break
-			@unknown default:
-				break
-			}
+	private var cornerRadius: CGFloat = 6
 
-			return NSMakeRect(x, y, width, height)
-		}
+	private static let borderWidth: CGFloat = 1
```

Comment:
and this one

Reply:
What is a `static let` if not a constant?

Reply:
derp, ignore